docs/cody/model-configuration/index.mdx

```diff
@@ -0,0 +1,553 @@
+| -------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `"sourcegraph"`      | [Cody Gateway](/cody/core-concepts/cody-gateway), offering access to various models from multiple services. |
+| `"openaicompatible"` | Any OpenAI-compatible API implementation.                                                                   |
+| `"awsBedrock"`       | [Amazon Bedrock](https://docs.aws.amazon.com/bedrock/)                                                      |
+| `"azureOpenAI"`      | [Microsoft Azure OpenAI](https://azure.microsoft.com/en-us/products/ai-services/openai-service/)            |
+| `"anthropic"`        | [Anthropic](https://www.anthropic.com)                                                                      |
+| `"fireworks"`        | [Fireworks AI](https://fireworks.ai)                                                                        |
+| `"google"`           | [Google Gemini](http://cloud.google.com/gemini) and [Vertex](https://cloud.google.com/vertex-ai/)           |
+| `"openai"`           | [OpenAI](http://platform.openai.com)                                                                        |
+| `"huggingface-tgi"`  | [Hugging Face Text Generation Interface](https://huggingface.co/docs/text-generation-inference/en/index)    |
+
+The configuration for `serverSideConfig` varies by provider type.
+
+For examples, refer to the [examples page](/cody/model-configuration/examples).
+
+## Model Overrides
+
+With a provider defined (either a Sourcegraph-supplied provider or a custom provider configured via the `providerOverrides`
+field), custom models can be specified for that provider by adding them to the `"modelConfiguration.modelOverrides"` section.
+
+This field is an array of items, each with the following fields:
+
+-   `modelRef` - Uniquely identifies the model within the provider namespace.
+    -   A string in the format `${providerId}::${apiVersionId}::${modelId}`.
+    -   To associate a model with your provider, `${providerId}` must match the provider’s ID.
+    -   `${modelId}` can be any URL-safe string.
+    -   `${apiVersionId}` specifies the API version, which helps detect compatibility issues between models and Sourcegraph instances.
+        For example, `"2023-06-01"` can indicate that the model uses that version of the Anthropic API. If unsure, you may set this to `"unknown"` when defining custom models.
+-   `displayName` - An optional, user-friendly name for the model. If not set, clients should display the `ModelID` part of the `modelRef` instead (not the `modelName`).
+-   `modelName` - A unique identifier used by the API provider to specify which model is being invoked.
+    This is the identifier that the LLM provider recognizes to determine the model you are calling.
+-   `capabilities` - A list of capabilities that the model supports. Supported values: "autocomplete" and "chat".
```

Comment:
When instance admins view the list of Sourcegraph-supplied models, they see more capabilities than listed here (“autocomplete” and “chat”). Additional capabilities include “edit” and “vision.” However, only “autocomplete,” “chat,” and “vision” are allowed values in the site config. Why is “edit” not included? Was this an oversight, or is it intentional? (See [PRIME-572](https://linear.app/sourcegraph/issue/PRIME-572/modelconfiguration-edit-capability-is-not-allowed-in-model-overrides) for more details.)

Also, it seems nothing prevents admins from designating a model with only the “chat” capability as an “autocomplete” model ([source](https://github.com/sourcegraph/sourcegraph/blob/4430c6d1f8bc1873611a2fc20508896100b98346/cmd/frontend/internal/modelconfig/builder.go#L239-L245)). What input do we actually expect from admins regarding capabilities?

cc: @slimsag

Reply:
Initially "autocomplete" and "chat" were the only supported capabilities. I think those are the most important, and wouldn't be worried about omitting the others from the docs here.

The others ("edit" and "vision") were added later (by Olaf I believe? I'm not sure but I didn't see the PR for it.) I think that those are more important for us to have ability to use in models.json ourselves, but maybe not important for admins to configure - though I'm not 100% sure on this. In any case, most Sourcegraph admins will not have access to those options yet since its only in the very latest release.

Reply:
> Also, it seems nothing prevents admins from designating a model with only the “chat” capability as an “autocomplete” model ([source](https://github.com/sourcegraph/sourcegraph/blob/4430c6d1f8bc1873611a2fc20508896100b98346/cmd/frontend/internal/modelconfig/builder.go#L239-L245)). What input do we actually expect from admins regarding capabilities?

If a model is declared as "chat", it will show up in the chat dropdown model selector of Cody. If it is listed as autocomplete, it can be used for autocomplete but we have no selector there so it's only whichever "default" model is configured.

It is possible to write that a model which is e.g. only good for autocomplete has "chat" functionality, and it just means if you go to Cody and type to it - it would give you a garbage response.